packages/platform-core/src/config/config_controller.ts

```diff
@@ -0,0 +1,130 @@
+import os from 'os';
+import path from 'path';
+import fs from 'fs';
+
+/**
+ * Used to interact with config file based on OS.
+ */
+export class ConfigController {
+  dirPath: string;
+  configFilePath: string;
+  _store: Record<string, unknown>;
+
+  /**
+   * Initializes common paths to telemetry configs.
+   */
+  constructor(
+    private readonly projectName = 'amplify',
+    private readonly configFileName = 'config.json'
+  ) {
+    this.dirPath = this.getConfigPath(this.projectName);
+    this.configFilePath = path.join(this.dirPath, this.configFileName);
+  }
+
+  /**
+   * Getter for cached config, retrieves config from disk if not cached already.
+   * If the store is not cached & config file does not exist, it will create a blank one.
+   */
+  private get store(): Record<string, unknown> {
+    if (this._store) {
+      return this._store;
+    }
+    if (fs.existsSync(this.configFilePath)) {
+      this._store = JSON.parse(fs.readFileSync(this.configFilePath).toString());
+    } else {
+      this._store = {};
+      this.write();
+    }
+    return this._store;
+  }
+
+  /**
+   * Creates project directory to store config if it doesn't exist yet.
+   */
+  private mkConfigDir() {
+    fs.mkdirSync(this.dirPath, { recursive: true });
+  }
+
+  /**
+   * Gets values from cached config by path.
+   */
+  get<T>(path: string) {
```

Comment:
instead of just casting to T, we could pass in an "expected type" of "number" | "string" | "boolean" (or maybe just boolean if that's all we need right now) and validate that the value is the expected type?